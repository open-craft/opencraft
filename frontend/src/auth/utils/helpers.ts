import { V2Api } from 'global/api';

export const checkAuthAndRefreshToken = async () => {
  const refreshToken = localStorage.getItem('token_refresh');
  let response = null;

  if (refreshToken) {
    try {
      response = await V2Api.authRefreshCreate({
        data: { refresh: refreshToken }
      });
    } catch (e) {
      return false;
    }
  }

  if (response) {
    localStorage.setItem('token_access', response.access);
    return true;
  }

  return false;
};
