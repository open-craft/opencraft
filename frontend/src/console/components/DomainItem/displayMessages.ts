const messages = {
  helpTitle: {
    defaultMessage: 'DNS Configuration',
    description: 'Title for DNS configuration help modal'
  },
  helpDescription: {
    defaultMessage:
      'This table shows you the DNS entries needed to connect your domain.' +
      ' Log in to your domain provider to manage these settings.',
    description: 'Description for DNS configuration help modal'
  }
};

export default messages;
