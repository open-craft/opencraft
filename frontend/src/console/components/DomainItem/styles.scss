@import "~styles/theme";
@import "~bootstrap/scss/functions";
@import "~bootstrap/scss/variables";
@import "~bootstrap/scss/mixins";
@import "~bootstrap/scss/utilities";

.list-item {
  margin: 20px 0px;
  font-size: 18px;

  &.color-green {
    color: $green;
  }

  .domain-type {
    color: $secondary-2;
    font-size: 14px;
  }

  .dns-config-icon {
    color: $alert-yellow-3;
  }

  .dns-config {
    color: $secondary-2;
    font-size: 16px;
    margin-left: 10px;
    border-bottom: 1px black solid;
  }

  .check-dns-btn {
    flex-grow: 1;
    text-decoration: none;
    appearance: none;
    border: 0;
    padding: 0;
    font-family: inherit;
    font-size: inherit;
    background-color: inherit;
  }
}

.options-dropdown-toggle {
  display: inline-flex;
  border-radius: 0.375rem;
  justify-content: center;
  background-color: white;
  color: black;
  font-size: 16px;
  padding: 0.5rem 1rem;

  .dropdown-toggle-caret {
    margin-left: 0.375rem;
  }
}

.options-dropdown-toggle:focus {
  border: blue solid 2px;
}

.options-dropdown-toggle::after {
  content: none;
}

.dns-config-help-modal{
  padding: 35px;

  .dns-config-help-modal-description {
    color: $dark-2;
    font-size: 16px;
    max-width: 716px
  }

  h2 {
    color: $dark-1;
    font-size: 30px;
    text-align: left;
    font-weight: 600;
    font-family: $playfair-font;
  }

  table {
    font-size: 16px;
  }

  button {
    font-size: 14px;
  }
}

.dns-config-help-modal-header{
  display: block;
  position: absolute;
  right: 0px;
  border: none;
  cursor: pointer;
  z-index: 10;
  padding-right: 25px;
}
