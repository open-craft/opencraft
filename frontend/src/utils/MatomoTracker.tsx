import React, { ReactNode } from 'react';
import {
  createInstance,
  MatomoProvider,
  useMatomo
} from '@datapunt/matomo-tracker-react';
import {
  MATOMO_ALIAS_DOMAIN,
  MATOMO_BASE_URL,
  MATOMO_MY_DOMAIN,
  MATOMO_SITE_ID
} from '../global/constants';

interface MatomoTrackerProps {
  children: ReactNode;
}

interface MatomoAdditionalConfiguration {
  [propName: string]: any;
}

interface MatomoConfiguration {
  urlBase: string;
  siteId: number;
  configurations?: MatomoAdditionalConfiguration;
}

interface MatomoUserIdTrackerProps {
  userId: string;
}

/*
 * This wrapper component initializes the Matomo tracking by wrapping the children
 * elements, if the tracking is enabled by providing a valid value for the 'MATOMO_BASE_URL'
 * constant. It allows tracking page views, events and the user id, when available.
 *
 * The 'MATOMO_ALIAS_DOMAIN' constant allows the cross-domain linking of another site with
 * this site so that they are treated as the same site in Matomo.
 */
export const MatomoTracker: React.FC<MatomoTrackerProps> = (
  props: MatomoTrackerProps
) => {
  let element = props.children;
  if (MATOMO_BASE_URL) {
    const matomoConfiguration: MatomoConfiguration = {
      urlBase: MATOMO_BASE_URL,
      siteId: parseInt(MATOMO_SITE_ID, 10)
    };
    if (MATOMO_ALIAS_DOMAIN) {
      const myDomain = `*.${MATOMO_MY_DOMAIN}`;
      const aliasDomain = `*.${MATOMO_ALIAS_DOMAIN}`;
      matomoConfiguration.configurations = {
        enableCrossDomainLinking: true,
        setDomains: [myDomain, aliasDomain]
      };
    }
    element = (
      <MatomoProvider value={createInstance(matomoConfiguration)}>
        {props.children}
      </MatomoProvider>
    );
  }
  return <>{element}</>;
};

/**
 * This is a component passes the provided user id to Matomo if Matomo tracking is enabled.
 * The 'pushInstruction' function exposed by the 'useMatomo()' hook is used to do this.
 *
 * This component is needed as all the components in the registration process are classes and hence
 * the 'yseMatomo' hook cannot be directly used in those components.
 */
export const MatomoUserIdTracker: React.FC<MatomoUserIdTrackerProps> = (
  props: MatomoUserIdTrackerProps
) => {
  const { pushInstruction } = useMatomo();
  if (MATOMO_BASE_URL && props.userId) {
    pushInstruction('setUserId', props.userId);
  }
  return null;
};
