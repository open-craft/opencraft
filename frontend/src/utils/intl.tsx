import React from "react";
import { FormattedMessage, MessageDescriptor } from "react-intl";

type MessageMap = { [k: string]: Omit<MessageDescriptor, 'id'> };

interface Props<T, K extends keyof T> {
    id: K;
    messages: T;
    values?: any;
}


export const WrappedMessage = (props: Props<MessageMap, string>) => {
    const {messages, id, ...values} = props;
    return (
        <FormattedMessage  {...messages[id]} id={id} {...values}/>
    );
};
