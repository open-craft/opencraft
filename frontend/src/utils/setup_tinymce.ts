import 'tinymce/tinymce';
import 'tinymce/themes/silver';
import 'tinymce/skins/ui/oxide/skin.min.css';
import 'tinymce/skins/ui/oxide/content.min.css';
import 'tinymce/plugins/advlist';
import 'tinymce/plugins/autolink';
import 'tinymce/plugins/lists';
import 'tinymce/plugins/link';
import 'tinymce/plugins/image';
import 'tinymce/plugins/charmap';
import 'tinymce/plugins/print';
import 'tinymce/plugins/preview';
import 'tinymce/plugins/anchor';
import 'tinymce/plugins/searchreplace';
import 'tinymce/plugins/visualblocks';
import 'tinymce/plugins/code';
import 'tinymce/plugins/fullscreen';
import 'tinymce/plugins/insertdatetime';
import 'tinymce/plugins/media';
import 'tinymce/plugins/table';
import 'tinymce/plugins/paste';
import 'tinymce/plugins/help';
import 'tinymce/plugins/wordcount';
