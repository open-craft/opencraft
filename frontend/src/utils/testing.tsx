import { createMemoryHistory } from 'history';
import * as React from 'react';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router';
import * as renderer from 'react-test-renderer';
import { applyMiddleware, createStore } from 'redux';
import thunk from 'redux-thunk';
import { createRootReducer } from '../global/reducers';

export const setupComponentForTesting = (
  reactContent: JSX.Element,
  storeContents = {}
) => {
  const middleware = applyMiddleware(thunk);

  const store = createStore(
    createRootReducer(createMemoryHistory()),
    storeContents,
    middleware
  );

  return renderer.create(
    <IntlProvider textComponent={React.Fragment} locale="en">
      <Provider store={store}>
        <MemoryRouter>{reactContent}</MemoryRouter>
      </Provider>
    </IntlProvider>
  );
};
