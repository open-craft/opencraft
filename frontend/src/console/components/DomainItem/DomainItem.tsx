import { EXTERNAL_DOMAIN_CNAME_VALUE } from 'global/constants';
import { OpenEdXInstanceConfigUpdateDnsConfigurationStateEnum as DnsStateEnum } from 'ocim-client';
import * as React from 'react';
import {
  Button,
  Col,
  Container,
  Dropdown,
  Modal,
  Row,
  Table
} from 'react-bootstrap';
import DropdownMenu from 'react-bootstrap/DropdownMenu';
import { WrappedMessage } from 'utils/intl';
import messages from './displayMessages';
import './styles.scss';

interface State {
  title: string;
  subtitle: string;
  // extra state to manage the empty title and subtitle and rendering
  renderBool: boolean;
}

interface Props {
  domainName?: string;
  isExternal: boolean;
  dnsState?: DnsStateEnum;
  onDelete?: Function;
}

interface DropdownButtonPropType {
  children?: React.ReactNode;
  onClick?: React.MouseEventHandler;
}

interface DomainConfigHelpProps {
  domainName?: string;
}

const DomainConfigHelp: React.FC<DomainConfigHelpProps> = (
  props: DomainConfigHelpProps
) => {
  const [show, setShow] = React.useState(false);

  return (
    <>
      <button
        type="button"
        className="d-flex flex-column justify-content-center align-items-center check-dns-btn"
        onClick={e => {
          e.preventDefault();
          setShow(true);
        }}
      >
        <div className="d-flex flex-row">
          <div className="dns-config-icon">
            <i className="fas fa-exclamation-triangle" />
          </div>
          <div className="dns-config">Check DNS Configuration</div>
        </div>
      </button>
      <Modal onHide={() => setShow(false)} show={show} centered size="lg">
        <Modal.Header className="dns-config-help-modal-header" closeButton />
        <Modal.Body>
          <Container className="dns-config-help-modal">
            <h2>
              <WrappedMessage messages={messages} id="helpTitle" />
            </h2>
            <Row>
              <Col className="dns-config-help-modal-description">
                <p>
                  <WrappedMessage messages={messages} id="helpDescription" />
                </p>
              </Col>
            </Row>
            <Table bordered>
              <thead>
                <tr>
                  <th>NAME</th>
                  <th>TYPE</th>
                  <th>VALUE</th>
                </tr>
              </thead>
              <tbody>
                <tr>
                  <td>{props.domainName}</td>
                  <td>CNAME</td>
                  <td>{EXTERNAL_DOMAIN_CNAME_VALUE}</td>
                </tr>
                <tr>
                  <td>
                    *.
                    {props.domainName}
                  </td>
                  <td>CNAME</td>
                  <td>{EXTERNAL_DOMAIN_CNAME_VALUE}</td>
                </tr>
              </tbody>
            </Table>
            <div className="d-flex flex-row">
              <div>
                <Button
                  size="lg"
                  className="addBtn"
                  variant="primary"
                  onClick={() => setShow(false)}
                >
                  I have made these changes
                </Button>
              </div>
            </div>
          </Container>
        </Modal.Body>
      </Modal>
    </>
  );
};

export class DomainListItem extends React.PureComponent<Props, State> {
  private getDropdown() {
    const customToggle = React.forwardRef<null, DropdownButtonPropType>(
      ({ children, onClick }, ref) => {
        return (
          <button
            type="button"
            className="options-dropdown-toggle d-flex flex-row"
            ref={ref}
            onClick={e => {
              e.preventDefault();
              if (onClick) {
                onClick(e);
              }
            }}
          >
            {children}
            <i className="dropdown-toggle-caret fas fa-angle-down" />
          </button>
        );
      }
    );

    return (
      <Dropdown className="d-flex flex-column align-items-center justify-content-center">
        <Dropdown.Toggle as={customToggle} id="domain-dropdown">
          <span>Options</span>
        </Dropdown.Toggle>

        <DropdownMenu alignRight>
          <Dropdown.Item
            onClick={(event: any) => {
              event.preventDefault();
              if (this.props.onDelete) {
                this.props.onDelete();
              }
            }}
          >
            Delete Domain
          </Dropdown.Item>
        </DropdownMenu>
      </Dropdown>
    );
  }

  public render() {
    // Determine the color of the domain name text based on DNS config status
    let domainColor: string = 'color-green';
    if (
      this.props.isExternal &&
      this.props.dnsState !== DnsStateEnum.Verified
    ) {
      domainColor = '';
    }

    return (
      <ul className="domain-list">
        <li className={`list-item ${domainColor}`}>
          <div className="d-flex flex-wrap justify-content-between">
            <div>
              <div>{this.props.domainName}</div>
              {this.props.isExternal ? (
                <div className="domain-type">Primary Domain</div>
              ) : (
                <div className="domain-type">Default Subdomain</div>
              )}
            </div>
            {this.props.isExternal &&
            this.props.dnsState !== DnsStateEnum.Verified ? (
              <DomainConfigHelp domainName={this.props.domainName} />
            ) : null}
            {this.props.isExternal ? this.getDropdown() : null}
          </div>
        </li>
      </ul>
    );
  }
}
