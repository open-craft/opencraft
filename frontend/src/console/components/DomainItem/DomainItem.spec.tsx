import React from 'react';
import { setupComponentForTesting } from "utils/testing";
import { OpenEdXInstanceConfigUpdateDnsConfigurationStateEnum as DnsStateEnum } from 'ocim-client';
import { DomainListItem } from './DomainItem';

it('renders correctly for subdomains', () => {
  const props = {
    domainName: 'test.opencraft.hosting',
    isExternal: false
  }
  const tree = setupComponentForTesting(
    <DomainListItem {...props}/>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
});

it('renders correctly for unverified external domains', () => {
  const props = {
    domainName: 'example.com',
    isExternal: true,
    dnsState: DnsStateEnum.Failed,
  }
  const tree = setupComponentForTesting(<DomainListItem {...props}/>).toJSON();
  expect(tree).toMatchSnapshot();
})

it('renders correctly for verified external domains', () => {
  const props = {
    domainName: 'example.com',
    isExternal: true,
    dnsState: DnsStateEnum.Verified,
  }
  const tree = setupComponentForTesting(<DomainListItem {...props}/>).toJSON();
  expect(tree).toMatchSnapshot();
})
