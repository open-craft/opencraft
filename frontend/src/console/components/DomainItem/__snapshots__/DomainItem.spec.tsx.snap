// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly for subdomains 1`] = `
<ul
  className="domain-list"
>
  <li
    className="list-item color-green"
  >
    <div
      className="d-flex flex-wrap justify-content-between"
    >
      <div>
        <div>
          test.opencraft.hosting
        </div>
        <div
          className="domain-type"
        >
          Default Subdomain
        </div>
      </div>
    </div>
  </li>
</ul>
`;

exports[`renders correctly for unverified external domains 1`] = `
<ul
  className="domain-list"
>
  <li
    className="list-item "
  >
    <div
      className="d-flex flex-wrap justify-content-between"
    >
      <div>
        <div>
          example.com
        </div>
        <div
          className="domain-type"
        >
          Primary Domain
        </div>
      </div>
      <button
        className="d-flex flex-column justify-content-center align-items-center check-dns-btn"
        onClick={[Function]}
        type="button"
      >
        <div
          className="d-flex flex-row"
        >
          <div
            className="dns-config-icon"
          >
            <i
              className="fas fa-exclamation-triangle"
            />
          </div>
          <div
            className="dns-config"
          >
            Check DNS Configuration
          </div>
        </div>
      </button>
      <div
        className="d-flex flex-column align-items-center justify-content-center dropdown"
        onKeyDown={[Function]}
      >
        <button
          className="options-dropdown-toggle d-flex flex-row"
          onClick={[Function]}
          type="button"
        >
          <span>
            Options
          </span>
          <i
            className="dropdown-toggle-caret fas fa-angle-down"
          />
        </button>
      </div>
    </div>
  </li>
</ul>
`;

exports[`renders correctly for verified external domains 1`] = `
<ul
  className="domain-list"
>
  <li
    className="list-item color-green"
  >
    <div
      className="d-flex flex-wrap justify-content-between"
    >
      <div>
        <div>
          example.com
        </div>
        <div
          className="domain-type"
        >
          Primary Domain
        </div>
      </div>
      <div
        className="d-flex flex-column align-items-center justify-content-center dropdown"
        onKeyDown={[Function]}
      >
        <button
          className="options-dropdown-toggle d-flex flex-row"
          onClick={[Function]}
          type="button"
        >
          <span>
            Options
          </span>
          <i
            className="dropdown-toggle-caret fas fa-angle-down"
          />
        </button>
      </div>
    </div>
  </li>
</ul>
`;
